src/components/DomainSearch/DomainSearch.tsx

```diff
@@ -0,0 +1,139 @@
+import type { FC } from 'react';
+
+import { useState } from 'react';
+import classNames from 'classnames/bind';
+import { Input, Button, LoadingIndicator } from '@zero-tech/zui/components';
+import { DEFAULT_NETWORK_PROTOCAL } from '../../constants/network';
+import { URLS } from '../../constants/urls';
+import { useBuyDomain, useDomainAvailability } from '../../hooks';
+import styles from './DomainSearch.module.scss';
+
+const cx = classNames.bind(styles);
+
+type DomainSearchTypes = 'default' | 'alternative' | 'nobutton';
+
+type DomainSearchProps = {
+	type?: DomainSearchTypes;
```

Comment:
Can we make this prop more descriptive? It took me a while to work out what `type` meant.